docs/react-types/ComponentProps.md

```diff
@@ -0,0 +1,57 @@
+---
+title: ComponentProps<T>
+---
+
+`ComponentProps<T>` is a utility type that lets you grab all valid props of an HTML or SVG element, or infer prop type for a component.
+
+## Parameters
+
+- `T extends keyof JSX.IntrinsicElements | JSXElementConstructor<any>`
+
+`keyof JSX.IntrinsicElements` is a set of all HTML and SVG elements, such as `"div"`, `"h1"` and `"path"`.
```

Comment:
This should be generalized to "host components". We want to move DOM typings to react-dom and then `React.JSX.IntrinsicElements` will only contain the host components your renderer specified.